src/key_exchange/tripledh.rs

```diff
@@ -387,44 +418,70 @@ fn derive_3dh_keys<KeyFormat: KeyPair, D: Hash>(
 
     let info: Vec<u8> = [
         STR_3DH,
-        &client_nonce,
-        &server_nonce,
-        &client_s_pk.to_arr(),
-        &server_s_pk.to_arr(),
+        &serialize(&client_nonce, 2),
+        &serialize(&server_nonce, 2),
+        &serialize(&client_s_pk.to_arr(), 2),
+        &serialize(&server_s_pk.to_arr(), 2),
     ]
     .concat();
 
-    const OUTPUT_SIZE: usize = 32;
-    let mut okm = [0u8; 3 * OUTPUT_SIZE];
-    let h = Hkdf::<D>::new(None, &ikm);
-    h.expand(&info, &mut okm)
-        .map_err(|_| InternalPakeError::HkdfError)?;
+    // TODO: could be made more efficient with re-using ikm
```

Comment:
Is it just a matter of `let initial_hash = {let mut h = D::new();
   ... foo.update(bar); ... h};` with a `derive_secrets(initial_secret: D, ...)` that
   uses a `mut initial_secret.clone()`?

Reply:
Thanks, yes you are right, I addressed this TODO in an update to this PR.